server/jetstream.go

```diff
@@ -276,41 +292,28 @@ func (s *Server) DisableJetStream() error {
 	s.setJetStreamDisabled()
 
 	if s.JetStreamIsClustered() {
-		wasLeader := s.JetStreamIsLeader()
+		isLeader := s.JetStreamIsLeader()
 		js, cc := s.getJetStreamCluster()
 		js.mu.RLock()
 		meta := cc.meta
 		js.mu.RUnlock()
 
-		s.transferRaftLeaders()
-
-		if wasLeader {
-			// Wait until the new metacontroller leader is established
-			const timeout = 2 * time.Second
-			maxWait := time.NewTimer(timeout)
-			defer maxWait.Stop()
-			t := time.NewTicker(50 * time.Millisecond)
-			defer t.Stop()
-		LOOP:
-			for {
+		if meta != nil {
+			if isLeader {
+				js.remapStreams(meta.ID())
+				s.Warnf("JetStream initiating meta leader transfer")
 				select {
 				case <-s.quitCh:
 					return nil
-				case <-maxWait.C:
-					break LOOP
-				case <-t.C:
-					js.mu.RLock()
-					isCurrent := cc.isCurrent()
-					js.mu.RUnlock()
-					if isCurrent {
-						break LOOP
-					}
+				case <-time.After(2 * time.Second):
```

Comment:
So it will always wait for 2secs here, there is no notification that it no longer needs to wait, right?

Reply:
Yes this is for the case when we are the metaleader and are going, so arbitrary to wait for the remappings to be distributed and committed.

Reply:
BTW this is optimistic, meaning the system is operational if we fail to remap on our way out the door.